alerts/config.go

```diff
@@ -5,116 +5,168 @@ package alerts
+// UrgentLowAlert extends Base with a threshold.
+type UrgentLowAlert struct {
+	Base `bson:",inline"`
 	// Threshold is compared the current value to determine if an alert should
 	// be triggered.
 	Threshold `json:"threshold"`
 }
 
-func (t ThresholdMixin) Validate(validator structure.Validator) {
-	t.Threshold.Validate(validator)
+func (a UrgentLowAlert) Validate(validator structure.Validator) {
+	a.Base.Validate(validator)
+	a.Threshold.Validate(validator)
+}
+
+// NotLoopingAlert extends Base with a delay.
+type NotLoopingAlert struct {
+	Base  `bson:",inline"`
+	Delay DurationMinutes `json:"delay,omitempty"`
 }
 
-// WithThreshold extends Base with ThresholdMixin.
-type WithThreshold struct {
-	Base           `bson:",inline"`
-	ThresholdMixin `bson:",inline"`
+func (a NotLoopingAlert) Validate(validator structure.Validator) {
+	a.Base.Validate(validator)
+	dur := a.Delay.Duration()
+	validator.Duration("delay", &dur).InRange(0, 2*time.Hour)
 }
 
-func (d WithThreshold) Validate(validator structure.Validator) {
-	d.Base.Validate(validator)
-	d.ThresholdMixin.Validate(validator)
+// NoCommunicationAlert extends Base with a delay.
+type NoCommunicationAlert struct {
+	Base  `bson:",inline"`
+	Delay DurationMinutes `json:"delay,omitempty"`
 }
 
-// WithDelay extends Base with DelayMixin.
-type WithDelay struct {
-	Base       `bson:",inline"`
-	DelayMixin `bson:",inline"`
+func (a NoCommunicationAlert) Validate(validator structure.Validator) {
+	a.Base.Validate(validator)
+	dur := a.Delay.Duration()
+	validator.Duration("delay", &dur).InRange(0, 6*time.Hour)
+}
+
+// LowAlert extends Base with threshold and a delay.
+type LowAlert struct {
+	Base `bson:",inline"`
+	// Threshold is compared the current value to determine if an alert should
+	// be triggered.
+	Threshold `json:"threshold"`
+	Delay     DurationMinutes `json:"delay,omitempty"`
 }
 
-func (d WithDelay) Validate(validator structure.Validator) {
-	d.Base.Validate(validator)
-	d.DelayMixin.Validate(validator)
+func (a LowAlert) Validate(validator structure.Validator) {
+	a.Base.Validate(validator)
+	dur := a.Delay.Duration()
+	validator.Duration("delay", &dur).InRange(0, 2*time.Hour)
+	a.Threshold.Validate(validator)
 }
 
-// WithDelayAndThreshold extends Base with both DelayMixin and ThresholdMixin.
-type WithDelayAndThreshold struct {
-	Base           `bson:",inline"`
-	DelayMixin     `bson:",inline"`
-	ThresholdMixin `bson:",inline"`
+// HighAlert extends Base with a threshold and a delay.
+type HighAlert struct {
+	Base `bson:",inline"`
+	// Threshold is compared the current value to determine if an alert should
+	// be triggered.
+	Threshold `json:"threshold"`
+	Delay     DurationMinutes `json:"delay,omitempty"`
 }
 
-func (d WithDelayAndThreshold) Validate(validator structure.Validator) {
-	d.Base.Validate(validator)
-	d.DelayMixin.Validate(validator)
-	d.ThresholdMixin.Validate(validator)
+func (a HighAlert) Validate(validator structure.Validator) {
```

Comment:
Just pointing out that a number of struct functions don't have the lowercased first letter of the struct name as the receiver name. (You indicated that an earlier one was a typo.) Not a big deal, but just making you aware. Okay if you don't want to change them now. (If you do, though, go through all of the structs in the PR and update them all.)

Reply:
Oh, my pattern is to use the first letter of the noun. So:

```
func (c *httpClient) Get() {}
```

So the `a` there is intentional on my part.